### DISCUSSION TITLE
Can the new Sphinx-Needs VsCode IDE use the autogenerated needs.json?

### DISCUSSION THREAD

**Discussion by twodrops** [useblocks/sphinx-needs#643] [Q&A]
OpenNeeds IDE (prior to Sphinx-Needs 1.0.1 Integration) generated an own `needs.json` under `needs/needs.json`.

```
needs_build_json = True
```
in `conf.py` generates `needs.json` directly under the build folder.

Can Sphinx-Needs-IDE be configured to use the last mentioned `needs.json`, so that it is not generated twice?
Can combining them corrupt the ".doctree" cache?

**Comment by danwos** [comment]
I think this is related to #633.

**Reply by twodrops** [reply]
Yes, I guess it is related.
Once `esbonio.sphinx.buildDir` is considered as part of #633, will Sphinx-Needs Idee consider the https://sphinxcontrib-needs.readthedocs.io/en/latest/configuration.html#needs-file configuration or always generate under `{esbonio.sphinx.buildDir}/needs/needs.json`

**Reply by twodrops** [reply]
@danwos @haiyangToAI When I checked the documentation I saw that Sphinx-Needs IDE expects needs.json to be generated with one of the two ways https://sphinxcontrib-needs.readthedocs.io/en/latest/ide/index.html#usage

We had a feature in Open-Needs IDE that the needs.json is created automatically upon save in the editor. Is this feature not supported anymore? If yes, where will the needs.json be created. It is not so clear from the documentation.
